### ISSUE TITLE
Cross-compiling from aarch64 to x86_64, error: unknown target 'x86_64'

### ISSUE THREAD

**Issue by deze333** [swiftlang/swift-sdk-generator#166]
Using M4 macOS as a host. Followed the instructions and successfully built two Linux SDKs:

```bash
> swift sdk list
6.0.2-RELEASE_ubuntu_jammy_aarch64
6.0.2-RELEASE_ubuntu_jammy_x86_64
```

Next created an example project `cross-compilation-test`, just as described in the docs.

Compiling on Apple Silicon for Linux with same architecture works:

```bash
swift build --swift-sdk 6.0.2-RELEASE_ubuntu_jammy_aarch64
```

Compiling on Apple Silicon for Linux with Intel architecture fails:

```bash
swift build --swift-sdk 6.0.2-RELEASE_ubuntu_jammy_x86_64

Building for debugging...
error: unknown target 'x86_64'
```

**Comment by xtremekforever** [comment]
Curious...if you use the latest main to generate a 6.0.3 Swift SDK now, do you get the same error?

**Comment by deze333** [comment]
Same error. I wonder if it's even possible to compile across processor architectures? Perhaps Swift compiler is only built to support single architecture target.

**Comment by euanh** [comment]
> I have generated the SDK with
> 
> swift run swift-sdk-generator make-linux-sdk --target x86_64

Please try:
```
swift run swift-sdk-generator make-linux-sdk --target x86_64-unknown-linux-gnu
```

Note the full target triple in this command.  With `x86_64` I see the same problem as you.

**Comment by xtremekforever** [comment]
> Note the full target triple of `x86_64-unknown-linux-gnu` in this command. With just `x86_64` I see the same problem as you.

This makes me wonder if we shouldn't put in a validation to make sure that the whole triple is provided to `--target` since if you don't it'll just use whatever you pass instead, which is invalid. We also have `--target-arch` to just specify the target architecture that defaults to `*-unknown-linux-gnu`, so maybe `--target` should enforce the whole triple.

**Comment by euanh** [comment]
> The full triple SDK build **almost** works... until the error message about missing`"swift-autolink-extract"`. I am on macOS.

Which version of macOS are you running, and which version of Xcode do you have installed?

**Comment by deze333** [comment]
> Seems like this is what most people are doing on macOS. There is the `--host-toolchain` param that can be added to include the open source toolchain in the Swift SDK:
> 
> `swift run swift-sdk-generator make-linux-sdk --target x86_64-unknown-linux-gnu --host-toolchain`
> 
> So then you don't have to switch between versions in Xcode. But, that's up to you on how you want to set things up.

Yes, I can see that it now downloads the whole open source toolchain and indeed successfully compiles the project. But it also spits out LOTS of duplicate library warnings found between newly installed open source toolchain and Xcode toolchain:

```bash
> swift build --swift-sdk 6.0.3-RELEASE_ubuntu_jammy_x86_64

objc[59766]: Class _TtC19CompilerSwiftSyntax16BumpPtrAllocator is implemented in both /Users/XXX/Library/org.swift.swiftpm/swift-sdks/6.0.3-RELEASE_ubuntu_jammy_x86_64.artifactbundle/6.0.3-RELEASE_ubuntu_jammy_x86_64/x86_64-unknown-linux-gnu/swift.xctoolchain/usr/lib/swift/host/libSwiftSyntax.dylib (0x1193e5138) and /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/libSwiftSyntax.dylib (0x117d52138). One of the two will be used. Which one is undefined.

objc[59766]: Class _TtC19CompilerSwiftSyntax23SourceLocationConverter is implemented in both /Users/XXX/Library/org.swift.swiftpm/swift-sdks/6.0.3-RELEASE_ubuntu_jammy_x86_64.artifactbundle/6.0.3-RELEASE_ubuntu_jammy_x86_64/x86_64-unknown-linux-gnu/swift.xctoolchain/usr/lib/swift/host/libSwiftSyntax.dylib (0x1193e5350) and /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/libSwiftSyntax.dylib (0x117d52350). One of the two will be used. Which one is undefined.

# and so on...
```

Any tips on how to avoid this?

**Comment by xtremekforever** [comment]
I'm not sure there's a workaround for that, but it seems like undesired behavior to me. I also see that happening on my macOS when I include the host toolchain in the Swift SDK, but am using the open-source toolchain with Swiftly.

This does not happen on Linux hosts, so I think it's probably a bug or non-optimal behavior. @MaxDesiatov should we file a bug for this somewhere (maybe SwiftPM project) to start tracking a fix for it?

**Comment by xtremekforever** [comment]
@deze333 would you say your original issue is now fixed? We updated the generator to default to including the --host-toolchain by default on macOS, to help out with newcomers. The README.md is also updated to reflect this.

**Comment by xtremekforever** [comment]
> I have generated the SDK with
> 
> swift run swift-sdk-generator make-linux-sdk --target x86_64
> Then
> 
> swift sdk install /Users/XYZ/swift-sdk-generator/Bundles/6.0.3-RELEASE_ubuntu_jammy_x86_64.artifactbundle
> 
> swift sdk list
> Build a project
> 
> # created minimal project as per docs
> cd ../cross-compilation-test 
> 
> swift build --swift-sdk 6.0.3-RELEASE_ubuntu_jammy_x86_64
> 
> # Error:
> Building for debugging...
> error: unknown target 'x86_64'

Looks like the target issue is the bug I filed where you can pass just the arch part of the target triple to `--target` and it lets you: #239
